README.md

```diff
@@ -21,27 +31,167 @@ The new HTTP Archive data pipeline built entirely on GCP
+            Workflows->>Cloud Storage: check if HAR manifest exists
+        end
+        Workflows->>Cloud Build: trigger job
+        Cloud Build->>Cloud Build: list HAR files and generate manifest file
+        Cloud Build->>Cloud Storage: upload HAR manifest to GCS
+    end
+    rect rgb(191, 223, 255)
+        Note right of Workflows: check BigQuery and run Dataflow jobs
+        break when BigQuery records exist for table and date
+            Workflows->>BigQuery: check all/combined tables for records in the given date
+        end
+        loop run jobs until retry limit is reached
+            Workflows->>Dataflow: run flex template
+            loop until job is complete
+                Workflows-->Dataflow: wait for job completion
+            end
+        end
+    end
+```
+
+### Development workflow
+
+```mermaid
+sequenceDiagram
+    autonumber
+    actor developer
+    participant Local as Local Environment / IDE
+    participant Dataflow
+    participant Cloud Build
+    participant Workflows
+
+    developer->>Local: create/update Dataflow code
+    developer->>Local: run Dataflow job with DirectRunner via run_*.py
+    developer->>Dataflow: run Dataflow job with DataflowRunner via run_pipeline_*.sh
+    developer->>Cloud Build: run build_flex_template.sh
+    developer->>Workflows: update flexTemplateBuildTag
+```
+
+### Manually running the pipeline
+
+```mermaid
+sequenceDiagram
+    actor developer
+    participant Local as Local Environment / IDE
+    participant Dataflow
+    participant PubSub
+    participant Workflows
+
+    alt run Dataflow job from local environment using the Dataflow runner
+        developer->>Local: clone repository and execute run_pipeline_*.sh
+    else run Dataflow job as a flex template
+        alt from local environment
+            developer->>Dataflow: clone repository and execute run_flex_template.sh
+        else from Google Cloud Console
+            developer->>Dataflow: use the Google Cloud Console to run a flex template as documented by GCP
+        end
+    else trigger a Google Workflows execution
+        alt
+            developer->>PubSub: create a new message containing a HAR manifest path from GCS
+        else
+            developer->>Workflows: rerun a previously failed Workflows execution
+        end
+    end
+```
 
 ## Run the pipeline
+Dataflow jobs can be triggered several ways:
+- Locally using bash scripts
+- From the Google Cloud Console
+- By publishing a Pub/Sub message
+
+### Locally using the `run_pipeline*.sh` scripts
+
+This method is best used when developing locally, as a convenience for running the pipeline's python scripts and GCP CLI commands.
 
 ```shell
-./run_pipeline_batch.sh
+# run the pipeline locally
+./run_pipeline_combined.sh
+./run_pipeline_all.sh
+
+# run the pipeline using a flex template
+./run_flex_template all [...]
+./run_flex_template combined [...]
+```
+
+### Running a flex template from the Cloud Console
+
+**TODO: ADD DETAILS**
```

Comment:
Would be good to capture the step-by-step instructions in the current PR

Reply:
Updated! Please have a look.